samples/simple_dgemv.cpp

```diff
@@ -0,0 +1,273 @@
+
+    const size_t bytesA = matrixA.size() * sizeof(float64_t);
+    const size_t bytesB = matrixB.size() * sizeof(float64_t);
+    const size_t bytesC = matrixC.size() * sizeof(float64_t);
+
+    CHECK_HIP_ERROR(hipMalloc(&d_a, bytesA));
+    CHECK_HIP_ERROR(hipMalloc(&d_b, bytesB));
+    CHECK_HIP_ERROR(hipMalloc(&d_c, bytesC));
+
+    CHECK_HIP_ERROR(hipMemcpy(d_a, matrixA.data(), bytesA, hipMemcpyHostToDevice));
+    CHECK_HIP_ERROR(hipMemcpy(d_b, matrixB.data(), bytesB, hipMemcpyHostToDevice));
+    CHECK_HIP_ERROR(hipMemcpy(d_c, matrixC.data(), bytesC, hipMemcpyHostToDevice));
+
+    auto blockDim = dim3(T_BLOCK_X, T_BLOCK_Y);
+    auto gridDim  = dim3(rocwmma::ceilDiv(m, ROCWMMA_M * T_BLOCK_X / WAVE_SIZE),
+                        rocwmma::ceilDiv(n, ROCWMMA_N * T_BLOCK_Y));
+
+    std::cout << "Launching dgemv kernel..." << std::endl;
+    hipEvent_t startEvent, stopEvent;
+    CHECK_HIP_ERROR(hipEventCreate(&startEvent));
+    CHECK_HIP_ERROR(hipEventCreate(&stopEvent));
+
+    hipExtLaunchKernelGGL(dgemv_rocwmma_d,
+                          gridDim,
+                          blockDim,
+                          0, // sharedMemBytes
+                          0, // stream
+                          startEvent, // Event start
+                          stopEvent, // event stop
+                          0, // flags
+                          m,
+                          n,
+                          k,
+                          d_a,
+                          d_b,
+                          d_c,
+                          d_c,
+                          lda,
+                          ldb,
+                          ldc,
+                          ldd,
+                          alpha,
+                          beta);
+
+    auto elapsedTimeMs = 0.0f;
+    CHECK_HIP_ERROR(hipEventSynchronize(stopEvent));
+    CHECK_HIP_ERROR(hipEventDestroy(startEvent));
+    CHECK_HIP_ERROR(hipEventDestroy(stopEvent));
+
+    // Echo performance
+    std::cout << "BlkM, BlkN, BlkK, "
+              << "MatM, MatN, MatK, " << std::endl;
+
+    std::cout << ROCWMMA_M << ", " << ROCWMMA_N << ", " << ROCWMMA_K << ", " << m << ", " << n
+              << ", " << k << std::endl;
+
+    std::cout << "Validating result with reference..." << std::endl;
+    // Bring kernel result back to host
+    std::vector<float64_t> matrixC_device(m * 1, std::numeric_limits<float64_t>::signaling_NaN());
+    CHECK_HIP_ERROR(hipMemcpy(matrixC_device.data(), d_c, bytesC, hipMemcpyDeviceToHost));
+
+    // Setup and run reference computation
+    std::vector<float64_t> matrixC_host(matrixC);
+    dgemv_cpu_h(m, n, k, matrixA.data(), matrixB.data(), matrixC_host.data(), alpha, beta);
+
+    compareEqual<float64_t>(matrixC_host.data(), matrixC_device.data(), m);
+
+    // Release device memory
+    CHECK_HIP_ERROR(hipFree(d_a));
+    CHECK_HIP_ERROR(hipFree(d_b));
+    CHECK_HIP_ERROR(hipFree(d_c));
+
+    std::cout << "Finished!" << std::endl;
+}
+
+int main()
```

Comment:
float64 gemm isn't supported on all architectures.
Can you add some protection here for this in main?

if(device->isGfx90a())
{   
    dgemv_test(...);
 }else{
    std::cout << "f64 not supported\n";
}